modules/local/appendmetadata/main.nf

```diff
@@ -0,0 +1,31 @@
+process APPEND_METADATA {
+    tag "append_metadata"
+    label 'process_low'
+
+    input:
+    val clusters_path  // cluster data as a TSV path
+                        // this needs to be "val", because "path"
+                        // won't stage the file correctly for exec
+    val metadata  // metadata to be appended, list of lists
+
+    output:
+    path("clusters_and_metadata.tsv"), emit: clusters
+
+    exec:
+    def merged = []
+    def rows
+
+    clusters_path.withReader { reader ->
+        rows = reader.readLines()*.split('\t')
+    }
+
+    for(int i = 0; i < rows.size(); i++)
+    {
+        merged.add(rows[i] + metadata[i])
```

Comment:
How is it that the metadata for a particular sample is being matched up to the appropriate row? Is the metadata list always in an identical order as the rows in the clusters_path file?

Reply:
I think the behaviour is that when it loads it in, it sets it up in the order it is in the samplesheet. However, I don't think this is guaranteed, so I made some changes to enforce this: 41556789d18f7d9c7e3e3dd741a97ce14886ed08